### PR TITLE
Fix pucket absorbtion

### PR DESCRIPTION
**Pull request by janstenpickle**
- now respects paritioning scheme
- modified tests to ensure fix holds

### CODE REVIEW

File: core/src/main/scala/com/intenthq/pucket/Pucket.scala [hunk 1]
```diff
@@ -99,16 +100,34 @@ trait Pucket[T] {
...
+
+
+  /** Target location for a given path under the current pucket
+    *
+    * @param pucket
+    * @param otherPath
+    * @return a fully qualified path under the current pucket, maintaining a partitioning scheme
+    */
+  def targetFile(pucket: Pucket[T], otherPath: Path): Path = {
+    val relative = new Path(StringUtils.removePattern(StringUtils.remove(otherPath.toString, pucket.path.toString), "^/"))
```
Comment (nathankleyn) [inline core/src/main/scala/com/intenthq/pucket/Pucket.scala]: I wasn't too sure about this, but I suppose it does the job. Just not sure about all the string replacement here and especially not the regex.

**Comment by nathankleyn** [comment]
:+1: 

Waiting on the build.

File: core/src/main/scala/com/intenthq/pucket/Pucket.scala (hunk 1 above)
Comment (janstenpickle) [inline core/src/main/scala/com/intenthq/pucket/Pucket.scala]: It's nasty I know, but hadoop doesn't really have any facility to do this.

Comment (nathankleyn) [inline core/src/main/scala/com/intenthq/pucket/Pucket.scala]: In fact - perhaps you could make this more efficient:

``` scala
StringUtils.removePattern(otherPath.toString, s"^${pucket.path.toString}/")
```

Presuming I've understood this correctly of course!

Comment (janstenpickle) [inline core/src/main/scala/com/intenthq/pucket/Pucket.scala]: Ah yeah that's better

File: core/src/main/scala/com/intenthq/pucket/Pucket.scala [hunk 2]
```diff
@@ -99,16 +100,34 @@ trait Pucket[T] {
...
+
+
+  /** Target location for a given path under the current pucket
+    *
+    * @param pucket
+    * @param otherPath
+    * @return a fully qualified path under the current pucket, maintaining a partitioning scheme
+    */
+  def targetFile(pucket: Pucket[T], otherPath: Path): Path = {
+    val relative = new Path(StringUtils.removePattern(otherPath.toString, s"^${pucket.path.toString}/"))
```
Comment (janstenpickle) [inline core/src/main/scala/com/intenthq/pucket/Pucket.scala:127]: Actually, this doesn't work. Hang on...

**Pull request merged**